### PR TITLE
Added Confirmation Modal for Delete Network

### LINKED ISSUE #6733: Additional confirmation when deleting custom RPC
https://www.figma.com/proto/cl052ly5iSXnqQqGU4V84s7A/Settings?node-id=66%3A1108&viewport=-1035%2C954%2C0.2857494652271271&scaling=min-zoom

When deleting custom RPC endpoint, a confirmation modal should pop up, requiring a second click before deleting the custom RPC endpoint.

### PR DESCRIPTION
**Pull request by akshitkrnagpal**
Fixes #6733

### CODE REVIEW

**Comment by akshitkrnagpal** [comment]
Hey

I have tried to add the confirmation modal using Figma as a reference. The changes might be complete but e2e test for delete network is failing right now because of the confirmation modal.

While I work on that it would be great if someone can take a look. 😄

File: ui/app/components/app/modals/confirm-delete-network/confirm-delete-network.component.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    return (
+      <Modal
+        onSubmit={this.handleDelete}
+        onCancel={() => this.props.hideModal()}
+        submitText={t('delete')}
+        cancelText={t('cancel')}
+        submitType="danger"
+      >
+        <ModalContent
+          title={`${t('deleteNetwork')}?`}
```
Comment (whymarrh) [inline ui/app/components/app/modals/confirm-delete-network/confirm-delete-network.component.js]: Can we move the question mark into the localized message?

**Review by whymarrh — CHANGES_REQUESTED** [review]
One small comment but otherwise this looks really good!

**Comment by akshitkrnagpal** [comment]
Hey @whymarrh 
I have updated the PR. 😄

**Review by whymarrh — APPROVED** [review]
LGTM, thanks @akshitkrnagpal!

**Pull request merged**